src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+const getUser = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxTodoId = [...todosFromServer].sort((a, b) => b.id - a.id)[0].id;
+
+const newTodo = (title: string, userId: number): Todo => ({
+  id: maxTodoId,
+  userId,
+  title,
+  completed: false,
+  user: getUser(userId),
+});
+
+const addTodo = (title: string, id: number) => {
+  maxTodoId += 1;
+  const todo = newTodo(title, id);
+
+  todos.push(todo);
+};
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const changeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const validatedTitle = value.replace(/[^A-Za-zА-ЯЁа-яё ]/, '');
+
+    setErrorTitle(false);
+    setTitle(validatedTitle);
+  };
+
+  const changeUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setErrorUser(false);
+    setUser(event.target.value);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const checkTitle = !title.trim();
+
+    if (checkTitle || !user) {
+      if (checkTitle) {
+        setErrorTitle(true);
+      }
+
+      if (!user) {
+        setErrorUser(true);
+      }
+
+      return;
+    }
```

Comment:
Try to avoid if inside if

Reply:
Added function for validation and fixed unnecessary if statements